src/main/kotlin/uk/gov/justice/digital/hmpps/gqlapi/resources/OffenderGraphqlController.kt

```diff
@@ -27,20 +34,39 @@ class OffenderGraphqlController(
 ) {
 
   @QueryMapping
-  fun offenders(): Flux<Offender> = offenderService.findAll()
+  fun offendersByLastName(@Argument lastName: String, env: DataFetchingEnvironment): Flux<Offender> =
+    offenderService.findByLastName(lastName).withToken(env)
 
   @QueryMapping
-  fun offendersByLastName(@Argument lastName: String): Flux<Offender> = offenderService.findByLastName(lastName)
-
-  @QueryMapping
-  fun offenderById(@Argument id: String): Mono<Offender> = offenderService.findById(id)
+  fun offenderById(@Argument id: String, env: DataFetchingEnvironment): Mono<Offender> =
+    offenderService.findById(id).withToken(env)
 
   @SchemaMapping
-  fun sentences(offender: Offender): Flux<Sentence> = offenderSentenceService.findByOffender(offender)
+  fun sentences(offender: Offender, env: DataFetchingEnvironment): Flux<Sentence> =
+    offenderSentenceService.findByOffender(offender).withToken(env)
 
   @SchemaMapping
-  fun offences(sentence: Sentence): Flux<Offence> = offenderSentenceOffencesService.findBySentence(sentence)
+  fun offences(sentence: Sentence, env: DataFetchingEnvironment): Flux<Offence> =
+    offenderSentenceOffencesService.findBySentence(sentence).withToken(env)
 
   @SchemaMapping
-  fun offenderManagers(offender: Offender): Flux<OffenderManager> = offenderManagerService.findByOffender(offender)
+  fun offenderManagers(offender: Offender, env: DataFetchingEnvironment): Flux<OffenderManager> =
+    offenderManagerService.findByOffender(offender).withToken(env)
 }
+
+@Configuration
+class AuthTokenExtractor : WebInterceptor {
```

Comment:
grab token header from GraphQL context that has a pointer the origin Web request 
(this might possibly done somewhere else without having the additional "env" variable 🤷🏽‍♂️)